use wazo-auth to finalize config

fredericlepied opened the pull request.

fredericlepied committed:
use wazo-auth to finalize config

fredericlepied committed:
inventories/uc-engine: expose only mandatory variables

sduthil commented on roles/engine-api-init/tasks/main.yml:
```diff
@@ -1,56 +1,4 @@
+- import_tasks: init.yml
+  when: validate_wazo_engine is not defined
+- import_tasks: validate.yml
+  when: validate_wazo_engine is defined
```
I don't like this pattern very much, as the engine-api-init role does two things. I'd prefer a separate validate role or playbook reusing the engine-api-init variables with `vars_file` or something, and avoid boolean variables.

fredericlepied replied:
I don't like it either but I don't know a good pattern to share variables between roles.

sduthil commented on roles/engine-api-init/tasks/validate.yml:
```diff
@@ -0,0 +1,31 @@
+- name: Get token
+  uri:
+    url: "https://{{ engine_api_host}}:{{ engine_api_port}}/api/auth/0.1/token"
```
Should we add a variable `engine_auth_path` like `engine_setupd_path` above? Or remove that kind of variable altogether?

fredericlepied replied:
like you want. I don't care ;-)

fredericlepied replied:
done with variables

sduthil commented on zuul.d/run.yaml:
```diff
@@ -16,3 +16,9 @@
+      become: yes
+      command: "ansible-playbook -i inventories/uc-engine test.yml"
+      args:
+        chdir: "{{ zuul.project.src_dir }}"
```
That's so nice and easy :D I like Zuul!

fredericlepied replied:
:-)

sduthil commented on roles/engine-api-init/tasks/init.yml:
```diff
@@ -0,0 +1,80 @@
+
+- name: Enable policy for API client
+  command: "wazo-auth-cli user add {{ api_client_name}} --policy api-client-policy"
+  when: engine_api_configure_wizard == "true"
```
I like this! How about a `block` + `when` instead of repeating `when`?

fredericlepied replied:
ok

fredericlepied replied:
done

fredericlepied committed:
take into account review comments:

- roles/engine-api-init/tasks/init.yml: use a block
- roles/engine-api-init/tasks/validate.yml: use variables for paths of services
